src/Module.cpp

```diff
@@ -319,9 +320,10 @@ void compile_multitarget(const std::string &fn_name,
     }
 
     std::vector<std::future<void>> futures;
-    // If we are running with HL_DEBUG_CODEGEN=1, choose a policy that enforces
+    // If we are running with HL_DEBUG_CODEGEN=1, use threads=1 to enforce
     // sequential execution, so that debug output won't be utterly incomprehensible
-    std::launch policy = debug::debug_level() > 0 ? std::launch::deferred : std::launch::async;
+    const size_t num_threads = (debug::debug_level() > 0) ? 1 : (targets.size() + 3);
```

Comment:
Spawning a thread for each task has other issues:
-- the number of std::thread instances you can create at one time is implementation-defined
-- if each thread does file I/O, this approach can easily run into the "too many files open" limit